nixos/modules/services/web-apps/pds.nix

```diff
@@ -0,0 +1,177 @@
+{
+  options.services.pds = {
+    enable = mkEnableOption "pds";
+
+    package = mkPackageOption pkgs "pds" { };
+
+    environment = mkOption {
+      type = types.attrsOf types.str;
+      default = { };
+      description = ''
+        Environment variables to set for the service. Secrets should be
+        specified using {option}`environmentFile`.
+      '';
+    };
+
+    environmentFiles = mkOption {
+      type = types.listOf types.path;
+      default = [ ];
+      description = ''
+        File to load environment variables from. Loaded variables override
+        values set in {option}`environment`.
+
+        Use it to set values of `PDS_JWT_SECRET`, `PDS_ADMIN_PASSWORD`,
+        and `PDS_PLC_ROTATION_KEY_K256_PRIVATE_KEY_HEX` secrets.
+        You can generate initial values with
+        ```
+        nix-build -A pds.passthru.generateSecrets
+        ./result/bin/generate-pds-secrets > secrets.env
+        ```
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 3000;
+      description = "Port to listen on";
+    };
+
+    hostname = mkOption {
+      type = types.str;
+      description = "Instance hostname";
+    };
+
+    dataDirectory = mkOption {
+      type = types.str;
+      default = "/var/lib/pds";
+      description = "Directory to store state";
+    };
+
+    blobstorageDirectory = mkOption {
+      type = types.nullOr types.str;
+      default = "/var/lib/pds/blocks";
+      description = "Directory to store blobstorage";
+    };
+
+    blobUploadLimit = mkOption {
+      type = types.int;
+      default = 52428800;
+      description = "Size limit of uploaded blobs";
+    };
+
+    plcUrl = mkOption {
+      type = types.str;
+      default = "https://plc.directory";
+      description = "URL of DID PLC directory";
+    };
+
+    bskyAppViewUrl = mkOption {
+      type = types.str;
+      default = "https://api.bsky.app";
+      description = "URL of bsky frontend";
+    };
+
+    bskyAppViewDid = mkOption {
+      type = types.str;
+      default = "did:web:api.bsky.app";
+      description = "DID of bsky frontend";
+    };
+
+    reportServiceUrl = mkOption {
+      type = types.str;
+      default = "https://mod.bsky.app";
+      description = "URL of mod service";
+    };
+
+    reportServiceDid = mkOption {
+      type = types.str;
+      default = "did:plc:ar7c4by46qjdydhdevvrndac";
+      description = "DID of mod service";
+    };
+
+    crawlers = mkOption {
+      type = types.str;
+      default = "https://bsky.network";
+      description = "URL of crawlers";
+    };
+
+    log.enable = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Enable logging";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.pds = {
+      description = "pds";
+      wants = [ "network-online.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      script = ''
+        exec ${getExe cfg.package}
```

Comment:
Could `ExecStart` be used instead since we're just calling the bare binary? That would remove the need to create an additional script derivation.